datastore/db/queries/inspection/__init__.py

```diff
@@ -204,64 +213,47 @@ def get_all_organization_inspection(cursor, org_id):
 
 
 def update_inspection(
-    cursor,
-    inspection_id,
-    verified=None,
-    label_info_id=None,
-    sample_id=None,
-    company_id=None,
-    manufacturer_id=None,
-    picture_set_id=None,
-    fertilizer_id=None,
-):
+    cursor: Cursor,
+    inspection_id: str | UUID,
+    user_id: str | UUID,
+    updated_data: Inspection,
+) -> Inspection:
     """
-    This function updates the inspection in the database.
+    Update inspection data in the database.
 
     Parameters:
-    - cursor (cursor): The cursor of the database.
-    - inspection_id (str): The UUID of the inspection.
-    - verified (boolean, optional): The value if the inspection has been verified by the user. Default is None.
-    - label_info_id (str, optional): The UUID of the label information. Default is None.
-    - sample_id (str, optional): The UUID of the sample. Default is None.
-    - company_id (str, optional): The UUID of the company. Default is None.
-    - manufacturer_id (str, optional): The UUID of the manufacturer. Default is None.
-    - picture_set_id (str, optional): The UUID of the picture set. Default is None.
-    - fertilizer_id (str, optional): The UUID of the fertilizer. Default is None.
+    - cursor (Cursor): Database cursor for executing queries.
+    - inspection_id (str | UUID): UUID of the inspection to update.
+    - user_id (str | UUID): UUID of the user performing the update.
+    - updated_data (Inspection): Updated inspection data as an Inspection object.
 
     Returns:
-    - The UUID of the inspection.
-    """
+    - Inspection: An Inspection object with updated inspection information.
 
+    Raises:
+    - InspectionUpdateError: Custom error for handling specific update issues.
+    """
     try:
-        query = """
-            UPDATE inspection
-            SET 
-                verified = COALESCE(%s, verified),
-                label_info_id = COALESCE(%s, label_info_id),
-                sample_id = COALESCE(%s, sample_id),
-                company_id = COALESCE(%s, company_id),
-                manufacturer_id = COALESCE(%s, manufacturer_id),
-                picture_set_id = COALESCE(%s, picture_set_id),
-                fertilizer_id = COALESCE(%s, fertilizer_id),
-                updated_at = CURRENT_TIMESTAMP
-            WHERE 
-                id = %s
-            RETURNING 
-                id
-            """
-        cursor.execute(
-            query,
-            (
-                verified,
-                label_info_id,
-                sample_id,
-                company_id,
-                manufacturer_id,
-                picture_set_id,
-                fertilizer_id,
-                inspection_id,
-            ),
-        )
-        return cursor.fetchone()[0]
+        updated_data_dict = updated_data.model_dump()
```

Comment:
Should we check if the parameter inspection_id and the inspection_id in the json are the same?

Reply:
It's done in the sql function.